src/payment/CardDetails.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Field } from 'redux-form';
+import { injectIntl, FormattedMessage } from '@edx/frontend-i18n';
+
+class CardDetails extends React.Component {
+  getNumericOptions(start, end) {
+    const items = [];
+    for (let i = start; i <= end; i += 1) {
+      items.push(<option key={i} value={i}>{i}</option>);
+    }
+    return items;
+  }
+
+  renderExpirationMonthOptions() {
+    return [
+      <option key="Month" value="">Month</option>,
+      ...this.getNumericOptions(1, 12),
+    ];
+  }
+
+  renderExpirationYearOptions() {
+    const currentYear = new Date().getFullYear();
+    return [
+      <option key="Year" value="">Year</option>,
+      ...this.getNumericOptions(currentYear, currentYear + 15),
+    ];
+  }
+
+  render() {
+    return (
+      <div className="mb-3">
+        <h2 className="section-heading">
+          <FormattedMessage
+            id="payment.card.details.billing.information.heading"
+            defaultMessage="Billing Information"
+            description="The heading for the credit card details billing information form"
+          />
+        </h2>
+
+        <div className="row">
+          <div className="col-lg-6 form-group">
+            <label htmlFor="cardNumber">
+              <FormattedMessage
+                id="payment.card.details.number.label"
+                defaultMessage="Card Number (required)"
+                description="The label for the required credit card number field"
+              />
+            </label>
+            <Field name="cardNumber" component="input" type="text" required className="form-control" />
```

Comment:
I was just noticing after adding react-axe - the fields need to have an `id` attribute for the labels' `htmlFor` to hook up to.  They don't actually point at `name` (which is what I thought for a while)